Bio/SeqUtils/__init__.py

```diff
@@ -667,6 +667,40 @@ def calculate(self, sequence):
 
         return exp(cai_value / cai_length)
 
+    def optimize(self, sequence, seq_type="DNA", table=standard_dna_table):
+        """Return a new DNA sequence with preferred codons only."""
+        try:  # If seq record is provided, convert to sequence
+            name = sequence.id
+            sequence = sequence.seq
+        except AttributeError:  # not a  SeqRecord object
+            pass
+        seq = sequence.upper()
+        # Make dict with amino acids referencing preferred codons
+        pref_codons = {aminoacid: [] for aminoacid in table.protein_alphabet}
+        for codon, aminoacid in table.forward_table.items():
+            if self[codon] == 1.0:
+                pref_codons[aminoacid] = codon
+        # Create amino acid sequence if DNA was provided
+        if seq_type == "DNA":
+            aa_seq = translate(seq)
+        elif seq_type == "protein":
+            aa_seq = seq
+        else:
+            raise TypeError
+        # Un-translate in loop using only preferred codons
+        optimized = Seq("")
```

Comment:
Creating a new ``Seq`` object in each loop iteration is an easily avoided overhead.

Work with strings and at the end you can do:

```python
return Seq(optimized)
```

Reply:
> Work with strings

or better yet, `bytes` objects; a string would require an extra encoding step.

Reply:
I have changed `optimized` to an empty list which is populated inside the loop and then transformed into a `seq` object before being returned. Does this resolve the problem pointed out here?

Reply:
This looks good. Michiel is probably right that using bytes would be even faster - provided the subsequences you are using are already available as bytes.